File: app/src/main/java/com/imangazalievm/bubbble/presentation/mvp/views/ShotDetailsView.java
package com.imangazalievm.bubbble.presentation.mvp.views;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.imangazalievm.bubbble.domain.models.Comment;
import com.imangazalievm.bubbble.domain.models.Shot;

import java.util.List;


@StateStrategyType(AddToEndSingleStrategy.class)
public interface ShotDetailsView extends MvpView {

    void showShot(Shot shot);
    void showLoadingProgress();
    void hideLoadingProgress();
    void showNoNetworkLayout();
    void hideNoNetworkLayout();
    void hideImageLoadingProgress();
    void showNewComments(List<Comment> newComments);
    void showCommentsLoadingProgress();
    void hideCommentsLoadingProgress();
    void showNoComments();
    void showImageSavedMessage();
    void showStorageAccessRationaleMessage();
    void showAllowStorageAccessMessage();
    void openAppSettingsScreen();
    @StateStrategyType(OneExecutionStateStrategy.class)
    void showShotSharing(String shotTitle, String shotUrl);
    @StateStrategyType(OneExecutionStateStrategy.class)
    void openInBrowser(String url);
    @StateStrategyType(OneExecutionStateStrategy.class)
    void openUserProfileScreen(long userId);
    @StateStrategyType(OneExecutionStateStrategy.class)
    void openShotImageScreen(Shot shot);

}

File: app/src/main/java/com/imangazalievm/bubbble/presentation/shotdetails/ShotDetailsView.kt
package com.imangazalievm.bubbble.presentation.shotdetails

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import com.imangazalievm.bubbble.domain.global.models.Comment
import com.imangazalievm.bubbble.domain.global.models.Shot
import com.imangazalievm.bubbble.presentation.global.mvp.BaseMvpView

@StateStrategyType(AddToEndSingleStrategy::class)
interface ShotDetailsView : BaseMvpView {

    fun showShot(shot: Shot)

    fun showLoadingProgress()

    fun hideLoadingProgress()

    fun showNoNetworkLayout()

    fun hideNoNetworkLayout()

    fun hideImageLoadingProgress()

    fun showNewComments(newComments: List<Comment>)

    fun showCommentsLoadingProgress()

    fun hideCommentsLoadingProgress()

    fun showNoComments()

    fun showImageSavedMessage()

    fun showStorageAccessRationaleMessage()

    fun showAllowStorageAccessMessage()

    fun openAppSettingsScreen()

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun showShotSharing(shotTitle: String, shotUrl: String)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun openInBrowser(url: String)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun openUserProfileScreen(userId: Long)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun openShotImageScreen(shot: Shot)

}